src/Sarif.Viewer.VisualStudio/SdkUiUtilities.cs

```diff
@@ -30,6 +33,9 @@ public static class SdkUiUtilities
         private static readonly Guid s_appIdUsesIsolatedCLR = new System.Guid("{074a44d3-1d9d-406c-9f91-c2a4982b1974}");
         internal static readonly Guid s_enviornmentThemeCategory = new System.Guid("624ed9c3-bdfd-41fa-96c3-7c824ea32e3d");
 
+        // Embedded link format: [link text](n) where n is a positive integer
+        private static string s_embeddedLinkRegexPattern = @"\[(?<link>[^\\\]]+)\]\((?<index>\d+)\)";
```

Comment:
`private const string EmbeddedLinkPattern;`

(I don't think you need "regex" in the name; it's clearly a regex pattern.)

Reply:
Fixed

Reply:
A couple of comments about the regex:

1. If you use the [`IgnorePatternWhitespace` option](https://docs.microsoft.com/en-us/dotnet/api/system.text.regularexpressions.regexoptions?view=netframework-4.7.1#System_Text_RegularExpressions_RegexOptions_IgnorePatternWhitespace), you can make this more understandable by splitting it up into multiple lines and adding comments:
```
private const string EmbeddedLinkPattern = @"
    \[               # An actual left bracket
    (?<link>
      [^\\\]]+       # One or more of anything that isn't a backslash and isn't a right bracket
    )
    \]               # An actual right bracket
    \(               # An actual left paren
    (?<index>
      \d+
    )
    \)               # An actual right paren
    ";
```

2. Having done that, I can now understand it well enough to ask: Why can't the link text contain a backslash?